src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -30,6 +30,20 @@ public sealed class XmlFormatReaderGenerator
     internal sealed class XmlFormatReaderGenerator
 #endif
     {
+        private static MethodInfo s_getUninitializedObjectMethodInfo;
+        private static Dictionary<Type, bool> s_typeHasDefaultConstructorMap;
+
+        static XmlFormatReaderGenerator()
```

Comment:
This explicit static cctor causes the C# compiler to not mark the type as beforefieldinit, which in turn means that the backend compiler needs to add checks to static methods to ensure that the type has been initialized. See #6016 for details.

To avoid this, consider removing the static cctor and initializing the static fields directly, e.g.

```csharp
private static readonly MethodInfo s_getUninitializedObjectMethodInfo =
    typeof(string)
    .GetTypeInfo()
    .Assembly
    .GetType("System.Runtime.Serialization.FormatterServices")
    ?.GetMethod("GetUninitializedObject", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static);

private static readonly Dictionary<Type, bool> s_typeHasDefaultConstructorMap = new Dictionary<Type, bool>();
```

Also, these static fields can be `readonly`.